test/resources/qaoa_utils.py

```diff
@@ -89,29 +89,25 @@ def initialize_params(cls, np_array):
 
     @classmethod
     def get_sgd_optimizer(cls, stepsize, params):
-        return tf.keras.optimizers.legacy.SGD(learning_rate=stepsize)
+        return tf.keras.optimizers.SGD(learning_rate=stepsize)
 
     @classmethod
     def convert_params_to_numpy(cls, params):
         return params.numpy()
 
     @classmethod
     def get_cost_and_step(cls, cost_function, params, optimizer):
-        def tf_cost():
-            global _cached_cost_before
-            _cached_cost_before = cost_function(params)
-            return _cached_cost_before
-
-        optimizer.minimize(tf_cost, params)
-        cost_before = _cached_cost_before
-
-        # Alternative:
-        # with tf.GradientTape() as tape:
-        #     cost_before = cost_function(params)
-        #
-        # gradients = tape.gradient(cost_before, params)
-        # optimizer.apply_gradients(((gradients, params),))
-
+        with tf.GradientTape() as tape:
```

Comment:
The later tests were failing in the pipeline but not in the integration builds. Were you able to work around this issue?